go/mysql/connection.go

```diff
@@ -47,5 +48,11 @@ func (this *ConnectionConfig) Equals(other *ConnectionConfig) bool {
 }
 
 func (this *ConnectionConfig) GetDBUri(databaseName string) string {
-	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", this.User, this.Password, this.Key.Hostname, this.Key.Port, databaseName)
+	var ip = net.ParseIP(this.Key.Hostname)
+	if (ip != nil) && (ip.To4() == nil) {
```

Comment:
`ParseIP` can return nil, but it's a typed `nil` that you can still call methods on.  Nil receivers are valid in Go.

https://play.golang.org/p/imTbiRbYmB